apps/hexagon_dma/process_rawinterleaved.cpp

```diff
@@ -0,0 +1,97 @@
+#include <stdio.h>
+#include <memory.h>
+#include <assert.h>
+#include <stdlib.h>
+#include "halide_benchmark.h"
+#include "pipeline_rawinterleaved.h"
+#include "HalideRuntimeHexagonDma.h"
+#include "HalideBuffer.h"
+#include "../../src/runtime/mini_hexagon_dma.h"
+
+int main(int argc, char **argv) {
+    if (argc < 3) {
+        printf("Usage: %s width height\n", argv[0]);
+        return 0;
+    }
+
+    const int width = atoi(argv[1]);
+    const int height = atoi(argv[2]);
+
+    // Fill the input buffer with random data. This is just a plain old memory buffer
+    const int buf_size = width * height * 4;
+    uint8_t *data_in = (uint8_t *)malloc(buf_size);
+    for (int i = 0; i < buf_size;  i++) {
+        data_in[i] = ((uint8_t)rand()) >> 1;
+    }
+
+
+    Halide::Runtime::Buffer<uint8_t> input(nullptr, width, height);
+ 
+    input.embed(2, 0);
```

Comment:
Just make the buffer with width, height, 4 above.

Also, you might be able to use Buffer::make_interleaved instead of the code below.

Reply:
Used Buffer::make_interleaved